### PR TITLE
feat(replays): Add looser type requirements for the user.id field

### PR DESCRIPTION
**Pull request by cmanallen**
Skip validation of the user.id field.  We will validate on the consumer side with the intent of adding stricter validation to relay in the future.

### CODE REVIEW

File: relay-replays/src/lib.rs [hunk 1]
```diff
@@ -147,7 +147,7 @@ struct VersionedMeta {
 
 #[derive(Debug, Default, Deserialize, Serialize)]
 struct User {
```
Comment (JoshFerge) [inline relay-replays/src/lib.rs:149]: can we just use this struct here instead? https://github.com/getsentry/relay/blob/14d431fa029a25916220f3e193bcc3a6bebbc6e1/relay-general/src/protocol/user.rs#L42

**Review by JoshFerge — APPROVED** [review]

File: relay-replays/src/lib.rs [hunk 2]
```diff
@@ -147,7 +147,7 @@ struct VersionedMeta {
...
-    id: Option<String>,
+    id: Option<Value>,
```
Comment (olksdr) [inline relay-replays/src/lib.rs:150]: Just a quick question: is User ID can be anything or is there an agreed upon definition of how the user should looks like?

**Review by olksdr — APPROVED** [review]

**Review by jjbayer — APPROVED** [review]
Please add a PR description.

File: relay-replays/src/lib.rs (hunk 2 above)
Comment (cmanallen) [reply relay-replays/src/lib.rs:150]: String or number.  Mostly I'm just not concerned with validating it here.  At least not yet. I want to allow anything for now and let the snuba consumer sort it out.

I want to use `relay_general::protocol::user::User` but it was a little difficult to integrate and its something I want to come back to.  If you have any tips for utilizing it let me know.

**Pull request merged**